sfcasts/crud-controller.md

```diff
@@ -1,171 +1,90 @@
+
+Pretty much everything about how our CRUD controller works is going to be controlled by overriding these configure methods that you see inside of here. We'll learn about all of these as we go along, but on a high level, `configureCrud` is here to help you configure the entire CRUD section. `configureAssets` allows you to control custom CSS and JavaScript in this section, `configureActions` allows you to control the actions you want, which are things like having an index page, having an edit action, or having a delete action. More on that later.
+
+The last important one is `configureFields`, which is going to configure which fields we see on both the index page and the form. But don't worry about those yet. We'll see all of them later.
+
+Then, below this, you can actually see all of the code that executes the pages. So `index` is our index page, `detail` is a detail show page, `edit` is the edit form.
+
+You can see the code that actually runs behind all of this, which is super useful when we're figuring out how to extend things. But... wait a second. If you scroll back up here to the configure methods, a few of these look familiar. Some of them *also* exist in the dashboard based controller class, specifically `configureAssets`, `configureCrud`, `configureActions`, and `configureFilters`. These all live side of `AbstractDashboardController`. There's `configureAssets`, and then further down here, CRUD actions and filters.
+
+So what's going on?
+
+Click "Questions" and look at the URL. It starts with `/admin` and then a bunch of query parameters. Turns out, everything EasyAdmin is handled via a single giant route. It all runs through the `DashboardController` route - the `/route` that's above the index. So when we go to the `QuestionCrudController`, it's actually matching this route here with extra query parameters to say which CRUD controller and which action runs. You can see `crudController` and `crudAction` right there.
+
+We're rendering `QuestionCrudController`, but in the context of our `DashboardController`. When we go to this page, in order to get the CRUD config, EasyAdmin first calls `configureCrud` on our dashboard. Then it calls `configureCrud` on our specific ride controllers. In this case, `QuestionCrudController` is super powerful. It means that we can override things on a dashboard level, like set some CRUD settings to apply to all sections or for one specific `crudController`.
+
+I'll prove it! Go back to `AbstractDashboardController`. In `configureCrud`, every area has four pages. I'll hold "cmd" and click to open this `Crud` thing. There's some constants on top here. Every CRUD section has an index page. That's this in 'edit' page, a 'new' page, and also a 'detail' page. Each page can then refer to a set of actions. These are links or buttons. For example, on the index page, right now, we have an action for editing and an action for deleting. We *could* add more actions or remove those actions.
```

Comment:
Does the "in" in the sentence "That's this in 'edit'..." sound right to you? I just caught that...